test/e2e/prometheus_test.go

```diff
@@ -1592,6 +1592,67 @@ func testInvalidRulesAreRejected(t *testing.T) {
 	}
 }
 
+func testPromReconcileStatusWhenInvalidRuleCreated(t *testing.T) {
+	t.Parallel()
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup(t)
+	ns := framework.CreateNamespace(context.Background(), t, testCtx)
+	framework.SetupPrometheusRBAC(context.Background(), t, testCtx, ns)
+	ruleFilesNamespaceSelector := map[string]string{"excludeFromWebhook": "true", "role": "rulefile"}
+
+	err := framework.AddLabelsToNamespace(context.Background(), ns, ruleFilesNamespaceSelector)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ruleName := "invalidrule"
+	alertName := "invalidalert"
+
+	_, err = framework.MakeAndCreateInvalidRule(context.Background(), ns, ruleName, alertName)
+	if err != nil {
+		t.Fatalf("expected invalid rule to be created in namespace %v", err)
+	}
+
+	prom := framework.MakeBasicPrometheus(ns, "basic-prometheus", "test-group", 1)
+	if _, err = framework.CreatePrometheusAndWaitUntilReady(context.Background(), ns, prom); err != nil {
+		t.Fatal("Creating prometheus failed: ", err)
+	}
+
+	var pollErr error
```

Comment:
ya correct. I just needed updated prometheus object after it becomes ready and `f.WaitForPrometheusReady` doesn't return updated object. Should we update that method to return updated prometheus since that will contain status as well but that will need change on other tests as well to update the method call.

For now I am getting the object again in this test to check for reconcile status

Reply:
my bad I didnt notice `WaitForResourceAvailable()` is called by `WaitForPrometheusReady()` which checks the reconciliation status. I have removed the extra check now